src/main/scala/mrpowers/jodie/DeltaHelpers.scala

```diff
@@ -125,6 +127,44 @@ object DeltaHelpers {
     }
   }
 
+  /**
+   * Gets the number of updated partitions in a Delta Table for a given time range. This is particularly useful in cases
+   * where you want to find partitions affected by append/merge operations and want to run a compaction or vacuum operation
+   * on them
+   */
+  def getUpdatedPartitions(path: String, startTime: Option[Instant] = None, endTime: Option[Instant] = None): Array[Map[String, String]] = {
+    DeltaLog
+      .forTable(SparkSession.active, path)
+      .snapshot
+      .allFiles
+      .filter(a => startTime.forall(a.modificationTime >= _.toEpochMilli) && endTime.forall(a.modificationTime <= _.toEpochMilli))
+      .withColumn("partitionValuesString", col("partitionValues").cast(StringType))
+      .dropDuplicates("partitionValuesString")
```

Comment:
cast to string for dropDuplicates as `dropDuplicates` directly on `partitionValues` column which have type Map will result in error
```
Cannot have map type columns in DataFrame which calls set operations(intersect, except, etc.), but the type of column partitionValues is map<string,string>;

```